docs/develop/scripting-fundamentals.md

```diff
@@ -218,6 +218,120 @@ function main(workbook: ExcelScript.Workbook) {
+
+![Explicit any message in editor](../images/explicit-any-editor-message.png)] 
+
+![Explicit any error](../images/explicit-any-error-message.png)] 
+
+In the above code `[5, 16] Explicit Any is not allowed` indicates that line # 5 column # 16 defines `any` type. This helps to locate the error line. 
+
+To get around this issue, always define the type of the variable. 
+
+If you are uncertain about the type of a variable, one cool trick in TypeScript allows you to define [union types](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html). This can be used for variables to hold range value, which can be only of many types. 
+
+```typescript
+   let value: (string | number | boolean);
+   // ^^ define this as a union type rather than any type ^^
+   value = someValue_from_another_source;
+   //...
+   someRange.setValue(value);
+```
+
+#### Type inference 
+
+In TypeScript, there are several places where [type inference](https://www.typescriptlang.org/docs/handbook/type-inference.html) is used to provide type information when there is no explicit type annotation. For example, in this code
+
+```typescript
+let x = 3;
+//  ^ = let x: number
+```
+
+The type of the x variable is inferred to be number. This kind of inference takes place when initializing variables and members, setting parameter default values, and determining function return types. 
+
+
+#### no-implicit-any rule
+A script requires the types of the variables used to be explicitly or implicity defined. If TypeScript compiler is unable to determine the type of a variable (either because type is not defined explicitly or type inference is not not possible), then you will receive a compilation time error (error prior to the running of the script).  You will see an error on the editor as well.
+
+![Implicit any error](../images/implicit-any-message.png)
+
+Following scripts will receive compilation time error as variables are defined without any types and TypeScript cannot determine the type at the time of declaration. 
+
+```typescript
+function main(workbook: ExcelScript.Workbook) {
+    let value; 
+    // ^^ the variable 'value' gets 'any' type as no type is defined. ^^
+    value = 10; 
+    // ^^Even when a numer type is assigned, the type of 'value' remains any. ^^
+    workbook.getActiveCell().setValue(value);
+    // ^^ Above line will fail because Office Scripts can't send an argument of type 'any' to Excel for processing. ^^
+    return;
+}
+```
+
+```typescript
+function main(workbook: ExcelScript.Workbook) {
+    let cell; 
+    // ^^ the variable 'cell' gets 'any' type as no type is defined. ^^
+    cell = workbook.getActiveCell().getValue();
+    console.log(cell.getValue());
+    // ^^ Office Scripts cannot assign Range type object to 'any' type variable. ^^    
+    return;
+}
+```
+
+To avoid this, use following instead. In each case, the type is declared at the time of declaration of variable. 
```

Comment:
Is this the preferred way, instead of casting in the method (i.e. `value as number`)? If so, we should rework the samples that broke after the original `any` Range signature change.